internal/service/static_ip/resource_static_ip.go

```diff
@@ -98,19 +97,35 @@ func resourceStaticIPCreate(ctx context.Context, d *schema.ResourceData, m inter
 	return resourceStaticIPRead(ctx, d, m)
 }
 
-func resourceStaticIPDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceStaticIPDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(*aiven.Client)
 
 	project, staticIPAddressId := schemautil.SplitResourceID2(d.Id())
 
-	err := client.StaticIPs.Delete(
+	staticIP, err := client.StaticIPs.Get(project, staticIPAddressId)
+	if err != nil && !aiven.IsNotFound(err) {
+		return diag.FromErr(err)
+	} else {
+		if staticIP.State == schemautil.StaticIpAssigned || staticIP.State == schemautil.StaticIpAvailable {
+			if err := client.StaticIPs.Dissociate(project, staticIPAddressId); err != nil {
```

Comment:
Let's say a service is created with a bunch of static IPs signed (each service has a minimum number, like for PG `startup-2`, it is two static IPs, and for `business-4`, it is 4 IPs). Then we want to disassociate one of these minimum required (these IPs have status `assigned`) IPs from the service API, which will retrieve `409 Static IP XXXX is in the state assigned`. 

So this code would only work if we created a PG service with the plan `startup-2` and added 3 IPs instead of 2 required (in this case first 2 IP addresses will in the state `assigned` and one in `available`)  and wanted to remove 3rd IP (that is in the state `available`). 

So at lest we should change `if staticIP.State == schemautil.StaticIpAssigned || staticIP.State == schemautil.StaticIpAvailable` to `if staticIP.State == schemautil.StaticIpAvailable`